docs/madr/decisions/057-rework-mesh-routes-matching.md

```diff
@@ -0,0 +1,902 @@
+where that part is represented by a real Kuma resource.
+
+In fact `targetRef` is the wrong concept for selecting workloads,
+since "targetRef refers to a workload" means we aren't targeting
+an arbitrary `Kind` of resource but instead always a `Dataplane`.
+
+In order to allow limiting the effects of a policy further
+we can introduce `spec.workloadSelector` which would basically be some variation of `labelSelector`.
+
+#### Advantages:
+
+- the workloads targeted by the policy is often "all workloads" or can be
+  inferred from the namespace so we kind of waste the top level, "special" `spec.targetRef`
+- tends to be less noisy as we move most interesting part of the policy from `to` /`from`section
+- Doesn't force every potential resource into a "to" or a "from", instead using the semantics of the `kind` itself
+	- `HTTPRoute` is a route, regardless of whether it's inbound or outbound
+	- `MeshService` _is_ a destination so `spec.targetRef.kind: MeshService` targets the configuration for a given destination
+	- `WorkloadIdentity` (obviously doesn't exist yet but we want to change how we manage identity)
+      would represent an identity so fits as `spec.targetRef` for e.g. `MeshTrafficPermission`
+
+#### Disadvantages:
+
+- `targetRef` is a single field, so it's not easy to share configuration
+  in a single resource to avoid duplication.
+  Though the very WIP policy attachment GEPs do propose a `targetRefs` field
+- requires a huge migration
+
+## Comparing UX of two approaches
+
+In both cases we assume that either `spec.targetRef` or `spec.workloadSelector` selects whole Mesh which can be omitted as default.
+With examples fist yaml is always with `to` section and second is with `spec.workloadSelector` approach.
+
+Looking at the examples we can see that when we omit selecting Mesh as default both approaches look almost the same. With 
+first approach giving a little bit more flexibility as `spec.to[]` is a list.
+
+### Examples
+
+1. Producer timeout to MeshService
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshTimeout
+metadata:
+  name: timeout-on-backend
+  namespace: backend-ns
+  labels:
+    kuma.io/mesh: default
+spec:
+  to:
+  - targetRef:
+      kind: MeshService
+      name: backend
+    default:
+      http:
+        requestTimeout: 10s
+```
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshTimeout
+metadata:
+  name: timeout-on-backend
+  namespace: backend-ns
+  labels:
+    kuma.io/mesh: default
+spec:
+  targetRef:
+    kind: MeshService
+    name: backend
+  default:
+    http:
+      requestTimeout: 10s
+```
+
+2. Consumer timeout to MeshService
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshTimeout
+metadata:
+  name: timeout-on-backend
+  namespace: frontend-ns
+  labels:
+    kuma.io/mesh: default
+spec:
+  to:
+  - targetRef:
+      kind: MeshService
+      name: backend
+      namespace: backend-ns
+    default:
+      http:
+        requestTimeout: 10s
+```
+
+```yaml
```

Comment:
I know that the MADR is about targeting `to` but would be nice to have an example that shows how to configure a Timeout on the inbound of the specific dataplane since there is no from section. Is there any change or should it work as it is?

Reply:
This would require a different resource

Reply:
I added a note about this to the option, we'd need a different policy `InboundTimeout` or `OutboundTimeout`